packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -508,16 +511,130 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
       controller: _optionsViewController,
       overlayChildBuilder: _buildOptionsView,
       child: TextFieldTapRegion(
-        child: SizedBox(
+        child: LayoutBuilder(
           key: _fieldKey,
-          child: Shortcuts(
-            shortcuts: _shortcuts,
-            child: Actions(
-              actions: _actionMap,
-              child: CompositedTransformTarget(
-                link: _optionsLayerLink,
-                child: fieldView,
+          builder: (BuildContext context, BoxConstraints constraints) {
+            // A post frame callback is used to get the field constraints so
+            // that the options view overlay is rebuilt when the field
+            // constraints change.
+            SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
+              if (!mounted) {
+                return;
+              }
+              _fieldBoxConstraints.value = constraints;
+            });
+            return Shortcuts(
+              shortcuts: _shortcuts,
+              child: Actions(
+                actions: _actionMap,
+                child: CompositedTransformTarget(
+                  link: _optionsLayerLink,
+                  child: fieldView,
+                ),
               ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+
+class _RawAutocompleteOptions extends StatefulWidget {
+  const _RawAutocompleteOptions({
+    required this.fieldKey,
+    required this.optionsLayerLink,
+    required this.optionsViewOpenDirection,
+    required this.overlayContext,
+    required this.textDirection,
+    required this.highlightIndexNotifier,
+    required this.builder,
+  });
+
+  final WidgetBuilder builder;
+  final GlobalKey fieldKey;
+
+  final LayerLink optionsLayerLink;
+  final OptionsViewOpenDirection optionsViewOpenDirection;
+  final BuildContext overlayContext;
+  final TextDirection? textDirection;
+  final ValueNotifier<int> highlightIndexNotifier;
+
+  @override
+  State<_RawAutocompleteOptions> createState() => _RawAutocompleteOptionsState();
+}
+
+class _RawAutocompleteOptionsState extends State<_RawAutocompleteOptions> {
+  late VoidCallback? removeCompositionCallback;
+  Offset fieldOffset = Offset.zero;
+
+  // Get the field offset if the field's position changes when its layer tree
+  // is composited, which occurs for example if the field is in a scroll view.
+  Offset _getFieldOffset() {
+    final RenderBox? fieldRenderBox = widget.fieldKey.currentContext?.findRenderObject() as RenderBox?;
+    final RenderBox? overlay = Overlay.of(widget.overlayContext).context.findRenderObject() as RenderBox?;
+    return fieldRenderBox?.localToGlobal(Offset.zero, ancestor: overlay) ?? Offset.zero;
+  }
+
+  void _onLeaderComposition(Layer leaderLayer) {
```

Comment:
Since we are already doing delayed layout using composite callbacks, can we remove the `CompositedTransformFollower`?

Reply:
I think so. You can compute the offset in the composition callback and adjust that in the next frame (I think this change is already doing that?).

Reply:
(following from offline) Keeping the `CompositedTransformFollower` since we want the options to strictly follow the field.